tests/interpreters/test_pdl_interpreter.py

```diff
@@ -473,3 +473,57 @@ def pdl_module():
 
     assert new_input_module_false.is_structurally_equivalent(ModuleOp([]))
     assert new_input_module_true.is_structurally_equivalent(input_module_true)
+
+
+def test_native_constraint_constant_parameter():
+    """
+    Check that `pdl.apply_native_constraint` can take constant attribute parameters
+    that are not otherwise matched.
+    """
+
+    @ModuleOp
+    @Builder.implicit_region
+    def input_module_true():
+        test.TestOp.create(properties={"attr": StringAttr("foo")})
+
+    @ModuleOp
+    @Builder.implicit_region
+    def input_module_false():
+        test.TestOp.create(properties={"attr": StringAttr("fooo")})
```

Comment:
```suggestion
        test.TestOp.create(properties={"attr": StringAttr("bar")})
```
🙏

Reply:
No, it has to be 4 letters sadly :(
I'll do "baar" if you want